pynio/block.py

```diff
@@ -37,6 +48,47 @@ def name(self):
     def type(self):
         return self._type
 
+    @property
+    def template(self):
+        return self._template
+
+    @template.setter
+    def template(self, value):
+        # do type checking later
+        self._template = value
+
+    @property
+    def config(self):
+        return self._config
+
+    @config.setter
+    def config(self, value):
+        if self._template is None:
+            self._config = value
+            return
+        config = deepcopy(self._template)
+        config.readonly = False
+        config.update(value, drop_unknown=True,
+                      drop_logger=self._instance.droplog)
+        self._config = config
+
+    def json(self):
+        return self._config.__basic__()
+
+    def _load_template(self, type, value, instance=None):
+        '''Set the template with a template gotten from nio'''
+        if instance is not None:
+            self._instance = instance
+        if self._instance is None:
+            raise TypeError("Block is not tied to an instance")
+
+        template = load_block(value)
```

Comment:
That won't work with how AttrDict handles things. It uses the class to recursively create the correct objects.

Interestingly, this may impact something that I wasn't thinking about before. I'll have to look into this more. However, I think loading functions are much cleaner and easier to manage than attaching __init__ functions onto a container class.